compiler/modules/front/const.bal

```diff
@@ -556,10 +583,14 @@ function expectsFloat(t:SemType? semType) returns boolean {
     }
 }
 
-function floatFromDecimalLiteral(string digits) returns float|error {
+function floatFromFpLiteral(string digits) returns float|error {
```

Comment:
This renaming is wrong, original name used `Decimal` to mean non-hex, not to mean floating point.

Reply:
Now we have `decimal`type, name is confusing, we should rename, or get rid of this. But we have used the term `Decimal` in `scan.bal` to mean 10-ary. @jclark what do you think about replacing `Decimal` with `Denary` in the code base when it means 10-ary.